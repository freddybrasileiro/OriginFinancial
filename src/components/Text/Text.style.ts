import styled from 'styled-components';

import {pxToRem} from '../../utils/pxToRem';
import {palette} from '../../theme/palette';

type TextProps = {
  variant: string,
};

export const Text = styled.span`
  font-family: 'Work Sans', sans-serif;
  font-style: normal;
  color: ${palette.general.darkBlue};
  font-weight: 500;
  ${(props: TextProps) => {
    if (props.variant === "label") {
      return `
        font-size: ${pxToRem(16)};
        line-height: ${pxToRem(20)};
      `
    } else if (props.variant === "body1") {
      return `
        font-weight: 600;
        font-size: ${pxToRem(20)};
        line-height: ${pxToRem(24)};
        letter-spacing: ${pxToRem(-0.166667)};
      `
    } else if (props.variant === "body2") {
      return `
        font-size: ${pxToRem(16)};
        line-height: ${pxToRem(20)};
      `
    } else if (props.variant === "body3") {
      return `
        font-size: ${pxToRem(18)};
        line-height: ${pxToRem(24)};
      `
    } else if (props.variant === "body4") {
      return `
        font-style: normal;
        font-size: ${pxToRem(40)};
        line-height: ${pxToRem(32)};
        letter-spacing: ${pxToRem(-0.266667)};
        color: ${palette.general.blue};
      `
    } else if (props.variant === "body5") {
      return `
        font-style: normal;
        font-size: ${pxToRem(12)};
        line-height: ${pxToRem(16)};
      `
    } else if (props.variant === "body6") {
      return `
        font-weight: 600;
        font-style: normal;
        font-size: ${pxToRem(12)};
        line-height: ${pxToRem(16)};
      `
    } else if (props.variant === "body7") {
      return `
        font-weight: 600;
        font-size: ${pxToRem(32)};
        line-height: ${pxToRem(40)};
        letter-spacing: ${pxToRem(-0.266667)};
      `
    } else if (props.variant === "body8") {
      return `
        font-size: ${pxToRem(16)};
        line-height: ${pxToRem(20)};
        color: ${palette.general.mediumGray};
      `
    }
    return '';
  }}
`;
