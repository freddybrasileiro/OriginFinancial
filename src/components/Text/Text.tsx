import React from 'react';

import * as S from './Text.style';

type TextProps = {
    children: any,
    variant: string,
};

const Text = ({ children, variant }: TextProps) => (
    <S.Text variant={variant} className={variant}>{children}</S.Text>
);

export default Text;