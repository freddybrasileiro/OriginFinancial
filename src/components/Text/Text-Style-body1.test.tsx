import * as React from 'react';
import { shallow } from 'enzyme';
import { Text } from './Text.style';

describe('Text', () => {
  it('Should render the Text style with variant body1', () => {
    expect(shallow(<Text variant="body1">My label</Text>)).toMatchSnapshot();
  });
});
