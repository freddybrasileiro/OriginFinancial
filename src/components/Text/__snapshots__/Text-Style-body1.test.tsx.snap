// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Should render the Text style with variant body1 1`] = `ShallowWrapper {}`;
